src/components/tree-view/TreeViewItem.tsx

```diff
@@ -51,51 +54,73 @@ export const TreeViewItem = <T,>({
     node.open();
   }, [hasLoadedChildren, loadChildren, node, isLeaf]);
 
+  const handleOver = useCallback(
+    (isOver: boolean) => {
+      if (isOver && !node.isOpen && !node.isDragging) {
+        timeoutRef.current = setTimeout(() => {
+          void handleClick();
+        }, 500);
+      }
+
+      if (timeoutRef.current && !isOver) {
+        clearTimeout(timeoutRef.current);
+      }
+    },
+    [handleClick, node.isOpen, node.isDragging]
+  );
+
   useEffect(() => {
-    if (node.willReceiveDrop && !node.isOpen) {
-      timeoutRef.current = setTimeout(() => {
-        void handleClick();
-      }, 200);
-    }
+    handleOver(isOver);
+  }, [isOver, handleOver]);
 
-    if (timeoutRef.current && !node.willReceiveDrop) {
-      clearTimeout(timeoutRef.current);
-    }
-  }, [node, handleClick]);
+  if (node.data.type === TreeViewNodeTypeEnum.SEPARATOR) {
+    return <div className="c__tree-view--node__separator" />;
+  }
+
+  if (node.data.type === TreeViewNodeTypeEnum.TITLE) {
+    return <div className="c__tree-view--node__title">{node.data.title}</div>;
+  }
 
   return (
-    <div
-      role="button"
-      onClick={onClick}
-      ref={dragHandle}
-      style={style}
-      className={clsx("c__tree-view--node", {
-        willReceiveDrop: node.willReceiveDrop,
-        selected: node.isSelected,
-      })}
+    <Droppable
```

Comment:
The "..." overflow appear too early, there is a lot of space left on the right, I know you did that because on hover the action button shows up, but I would be better to display the "..." only when the action button is displayed on hover

Maybe is it not directly related to the tree itself but the node rendering

<img width="428" alt="Capture d’écran 2025-03-06 à 15 24 29" src="https://github.com/user-attachments/assets/b866bd84-55e2-443e-bef6-e4377565fdd8" />